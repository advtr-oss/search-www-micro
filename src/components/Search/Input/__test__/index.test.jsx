/**
 * @jest-environment jsdom
 */

import React from 'react'
import user from '@testing-library/user-event'
import { render as _render, screen, act, waitFor } from '@testing-library/react'
import { ThemeProvider } from '@advtr/tidy'

import { Input } from '../index'

const render = (ui, options) =>
  _render(ui, { wrapper: ({ children }) => <ThemeProvider colorScheme='light'>{children}</ThemeProvider>, ...options })

describe('Input', function () {
  it('should display the base input', function () {
    const { container } = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display a disabled base input', function () {
    const { container } = render(<Input isDisabled />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should clear button click', async function () {
    user.setup()

    // Just to test `handleClick` was called
    const handleClick = jest.fn()

    const { getByLabelText } = render(<Input onClick={handleClick} />)

    await act(async () => {
      // Only works on the SVG
      const btn = getByLabelText('input-icon')
      await user.click(btn)
    })

    await waitFor(() => expect(handleClick).toBeCalledTimes(1))
  })

  it('should handle text input', async function () {
    user.setup()

    // Just to test `handleInput` was called
    const text = 'Hello World'
    const handleChange = jest.fn()

    render(<Input onChange={handleChange} />)

    const input = screen.getByLabelText('Search autocomplete')
    expect(input.value).toBe('')

    await act(async () => {
      await user.type(input, text)
    })

    expect(handleChange).toBeCalledTimes(text.length)
    expect(input).toHaveValue(text)
  })
})
