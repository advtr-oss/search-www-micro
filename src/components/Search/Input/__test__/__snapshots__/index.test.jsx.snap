// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should display a disabled base input 1`] = `
.c4 {
  width: 100%;
  border-style: none;
  color: #24292f;
  background-color: #ffffff;
}

.c4::-webkit-input-placeholder {
  color: #6e7781;
}

.c4::-moz-placeholder {
  color: #6e7781;
}

.c4:-ms-input-placeholder {
  color: #6e7781;
}

.c4::placeholder {
  color: #6e7781;
}

.c4:disabled {
  color: #57606a;
  background-color: #f6f8fa;
}

.c4:focus {
  outline: none;
}

.c1 {
  position: relative;
}

.c1 {
  background-color: #ffffff !important;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  padding: 0 8px;
  border-radius: 4px;
  width: 100%;
  height: 48px;
}

.c1:disabled {
  color: #57606a !important;
  background-color: #f6f8fa !important;
}

.c1 > .c3 {
  padding: 0 8px 0 8px;
  color: inherit;
  background-color: inherit;
  line-height: 40px;
  font-weight: 500;
}

.c2 {
  cursor: pointer;
  display: none;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 8px;
}

.c2 {
  border-style: none;
  border-radius: 10px;
  height: 40px;
}

.c2 {
  color: #24292f;
  background-color: inherit;
}

.c2:disabled {
  cursor: default;
}

.c2:disabled > svg {
  cursor: default;
  color: #57606a;
}

.c2 > svg {
  width: 16px;
  height: 16px;
}

.c2:enabled > svg:hover {
  color: #0969da;
}

.c0 button {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      aria-disabled="true"
      class="c2"
      disabled=""
    >
      <svg
        aria-disabled="true"
        aria-label="input-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </button>
    <input
      aria-disabled="true"
      aria-label="Search autocomplete"
      aria-readonly="false"
      autocomplete="off"
      class="c3 c4 input--disabled input--empty"
      disabled=""
      placeholder="Search the world..."
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input should display the base input 1`] = `
.c4 {
  width: 100%;
  border-style: none;
  color: #24292f;
  background-color: #ffffff;
}

.c4::-webkit-input-placeholder {
  color: #6e7781;
}

.c4::-moz-placeholder {
  color: #6e7781;
}

.c4:-ms-input-placeholder {
  color: #6e7781;
}

.c4::placeholder {
  color: #6e7781;
}

.c4:disabled {
  color: #57606a;
  background-color: #f6f8fa;
}

.c4:focus {
  outline: none;
}

.c1 {
  position: relative;
}

.c1 {
  background-color: #ffffff !important;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  padding: 0 8px;
  border-radius: 4px;
  width: 100%;
  height: 48px;
}

.c1:disabled {
  color: #57606a !important;
  background-color: #f6f8fa !important;
}

.c1 > .c3 {
  padding: 0 8px 0 8px;
  color: inherit;
  background-color: inherit;
  line-height: 40px;
  font-weight: 500;
}

.c2 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 8px;
}

.c2 {
  border-style: none;
  border-radius: 10px;
  height: 40px;
}

.c2 {
  color: #24292f;
  background-color: inherit;
}

.c2:disabled {
  cursor: default;
}

.c2:disabled > svg {
  cursor: default;
  color: #57606a;
}

.c2 > svg {
  width: 16px;
  height: 16px;
}

.c2:enabled > svg:hover {
  color: #0969da;
}

.c0 button {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2"
    >
      <svg
        aria-label="input-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </button>
    <input
      aria-disabled="false"
      aria-label="Search autocomplete"
      aria-readonly="false"
      autocomplete="off"
      class="c3 c4 input--empty"
      placeholder="Search the world..."
      type="text"
      value=""
    />
  </div>
</div>
`;
